File: VB/WebSite/App_Code/Objects.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Public Class User
	Inherits XPObject
	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Private _FullName As String
	<DisplayName("Full Name")> _
	Public Property FullName() As String
		Get
			Return _FullName
		End Get
		Set(ByVal value As String)
			SetPropertyValue("FullName", _FullName, value)
		End Set
	End Property

	<Association("Users-Roles")> _
	Public ReadOnly Property Roles() As XPCollection(Of Role)
		Get
			Dim result As XPCollection(Of Role) = GetCollection(Of Role)("Roles")
			result.DisplayableProperties = "Name"
			Return result
		End Get
	End Property
End Class

Public Class Role
	Inherits XPObject
	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Private _Name As String
	Public Property Name() As String
		Get
			Return _Name
		End Get
		Set(ByVal value As String)
			SetPropertyValue("Name", _Name, value)
		End Set
	End Property

	<Association("Users-Roles")> _
	Public ReadOnly Property Users() As XPCollection(Of User)
		Get
			Dim result As XPCollection(Of User) = GetCollection(Of User)("Users")
			result.DisplayableProperties = "FullName"
			Return result
		End Get
	End Property
End Class

File: CS/WebSite/App_Code/Objects.cs
using System;
using DevExpress.Xpo;

public class User : XPObject
{
    public User(Session session) : base(session) { }

    private string _FullName;
    [DisplayName("Full Name")]
    public string FullName
    {
        get
        {
            return _FullName;
        }
        set
        {
            SetPropertyValue("FullName", ref _FullName, value);
        }
    }

    [Association("Users-Roles")]
    public XPCollection<Role> Roles
    {
        get
        {
            XPCollection<Role> result = GetCollection<Role>("Roles");
            result.DisplayableProperties = "Name";
            return result;
        }
    }
}

public class Role : XPObject
{
    public Role(Session session) : base(session) { }

    private string _Name;
    public string Name
    {
        get
        {
            return _Name;
        }
        set
        {
            SetPropertyValue("Name", ref _Name, value);
        }
    }

    [Association("Users-Roles")]
    public XPCollection<User> Users
    {
        get
        {
            XPCollection<User> result = GetCollection<User>("Users");
            result.DisplayableProperties = "FullName";
            return result;
        }
    }
}

